JsonParser.cpp

```diff
@@ -0,0 +1,61 @@
+#include "JsonParser.h"
+#include <iostream>
+
+#include <fstream>
+#include <regex>
+#include <algorithm>
+
+std::map<std::string, std::any> JsonParser::ParseString(const std::string &input){
+    std::map<std::string, std::any> data;
+    std::smatch searchMatches;
+    
+    const std::regex searchRegex("\"([a-zA-Z0-9]+)\"\\s*:\\s*(\"[^\"]+\")?([0-9]*[.]?[0-9]+)?(true|false)?[,\n}]{1}");
+
+    std::string worker(input), key, value;
+
+    while (std::regex_search(worker, searchMatches, searchRegex))
+    {
+        key = searchMatches[1].str();
```

Comment:
És mi lesz, ha kifelejtik a kulcsot?

Reply:
@wajzy Abban az esetben nem találja meg a regex a sort, csak olyan sorokat talál, amik szintaktikailag helyesek.

Reply:
Ehh, hát igen, viszont a hibáról nem fogunk tudomást szerezni sem... Nem nagyon tetszik ez a megközelítés (mármint a hibák csendben átlépése), de végül így is működhet jól a program. A hibás sorhoz tartozó kulcs előbb-utóbb valahol hiányozni fog, és azt meg fel fogjuk fedezni, legfeljebb majd nem azt az üzenetet látja a user, hogy öreg, ezt elgépelted itt meg itt, hanem azt, hogy meg sem adtad. Kicsit félrevezető, de a végeredmény kb. ugyanaz.